### PR TITLE
I842 wait for seed word

### LINKED ISSUE #842: Use callback in placeSeedWord method
There's a race condition here, it should be corrected in `ui/app/actions.js`.

### PR DESCRIPTION
**Pull request by danfinlay**
When displaying seed words, we were not using a callback, which had some race condition potential.  This is simply a little cleaner and more correct.

Fixes #842

### CODE REVIEW

File: app/scripts/keyring-controller.js [hunk 1]
```diff
@@ -173,10 +173,15 @@ module.exports = class KeyringController extends EventEmitter {
     })
   }
 
-  placeSeedWords () {
+  placeSeedWords (cb) {
     const firstKeyring = this.keyrings[0]
     const seedWords = firstKeyring.serialize().mnemonic
     this.configManager.setSeedWords(seedWords)
+
+    if (cb && typeof cb === 'function') {
```
Comment (kumavis) [inline app/scripts/keyring-controller.js]: are we getting non-cb stuff being passed in?
this is a code smell, id like to remove it

Comment (kumavis) [inline app/scripts/keyring-controller.js]: also we're only emiting an update if theres a cb passed in

File: test/unit/account-link-test.js [hunk 2]
```diff
@@ -3,8 +3,8 @@ var linkGen = require('../../ui/lib/account-link')
 
 describe('account-link', function() {
 
-  it('adds testnet prefix to ropsten test network', function() {
-    var result = linkGen('account', '3')
+  it('adds testnet prefix to morden test network', function() {
+    var result = linkGen('account', '2')
```
Comment (kumavis) [inline test/unit/account-link-test.js]: why ropsten->morden?

**Review by kumavis — CHANGES_REQUESTED** [review]

File: app/scripts/keyring-controller.js (hunk 1 above)
Comment (flyswatter) [inline app/scripts/keyring-controller.js]: This method is sometimes called synchronously. Look at the code context to see why.

Comment (flyswatter) [inline app/scripts/keyring-controller.js]: We don't need to emit an update in the other case. We already emit the update after this call, after some other changes.

I guess we don't have to type check. Seems like safer coding, but I guess it doesn't matter.

**Comment by flyswatter** [comment]
Oh whoops, it was correct to not fix the etherscan link test in this PR.

File: app/scripts/keyring-controller.js [hunk 3]
```diff
@@ -173,10 +173,15 @@ module.exports = class KeyringController extends EventEmitter {
...
 
-  placeSeedWords () {
+  placeSeedWords (cb) {
     const firstKeyring = this.keyrings[0]
     const seedWords = firstKeyring.serialize().mnemonic
     this.configManager.setSeedWords(seedWords)
+
+    if (cb) {
+      cb()
+      this.emit('update')
```
Comment (kumavis) [inline app/scripts/keyring-controller.js]: pls move `this.emit('update')` outside the conditional

**Review by kumavis — CHANGES_REQUESTED** [review]

File: app/scripts/keyring-controller.js (hunk 3 above)
Comment (flyswatter) [inline app/scripts/keyring-controller.js]: Could you explain why? It seems like this is just adding update spam.  We already transmit an update in the synchronous usage of this method.

**Comment by flyswatter** [comment]
Seems like some PR got merged in the wrong order, and so we lost ropsten link support, it's been re-included in this PR.

File: app/scripts/keyring-controller.js (hunk 3 above)
Comment (kumavis) [inline app/scripts/keyring-controller.js]: > we already transmit an update in the synchronous usage of this method.

where? and if so, then we dont need to do it here at all
emiting updates is unrelated to whether or not the caller wants to be informed when we're done

Comment (flyswatter) [inline app/scripts/keyring-controller.js]: We only call it synchronously in `createNewVaultAndKeychain()`, which manages its own updates, which is why it needs to be called in the `cb` version and not in the sync version, but since that's hard to follow, I'll just move it.

**Review by kumavis — APPROVED** [review]
